ledger/participant-integration-api/src/main/scala/platform/apiserver/RateLimitingInterceptor.scala

```diff
@@ -133,26 +162,53 @@ private[apiserver] final class RateLimitingInterceptor(
       count: InstrumentedCount,
       limit: Int,
       fullMethodName: String,
-  ): Either[DamlError, Unit] = {
+  ): LimitResult = {
     val queued = count.queueSize
     if (queued > limit) {
-      val damlError = ThreadpoolOverloaded.Rejection(
-        name = count.name,
-        queued = queued,
-        limit = limit,
-        metricPrefix = count.prefix,
-        fullMethodName = fullMethodName,
+      OverLimit(
+        ThreadpoolOverloaded.Rejection(
+          name = count.name,
+          queued = queued,
+          limit = limit,
+          metricPrefix = count.prefix,
+          fullMethodName = fullMethodName,
+        )
+      )
+    } else UnderLimit
+  }
+
+  private def streamsUnderLimit(fullMethodName: String): LimitResult = {
+    if (activeStreamsCounter.getCount >= config.maxStreams) {
```

Comment:
if activeStreamsCounter is == maxStreams, it should still be fine no?

Reply:
But then this also really limits the max number of subscriptions to maxStreams - 1, potentially confusing users, no?

Reply:
The rate limiting test is applied before the count (for the stream being created) is incremented.

At the point the `maxStream`'th stream is being created `activeStreamsCounter` will contain `maxStream-1` (and so not be limited).  (the counter is incremented post rate limiting check).